src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultLSPDocumentSynchronizer.cs

```diff
@@ -132,6 +138,17 @@ internal void DocumentManager_Changed(object sender, LSPDocumentChangeEventArgs
                         _virtualDocumentContexts.Remove(virtualDocument.Uri);
                     }
                 }
+                else if (args.Kind == LSPDocumentChangeKind.VirtualDocumentChanged)
+                {
+                    if (args.VirtualOld.Snapshot.Version == args.VirtualNew.Snapshot.Version)
```

Comment:
@NTaylorMullen, if we remove this condition, can we just remove the need for `VirtualDocumentBuffer_PostChanged`?

Reply:
No, in the case that there were actual changes we'll notify "synchronized" before embedded language servers have had the chance to see the changes due to the way TextBuffer's Change events fire (it happens asynchronously). Meaning we'd notify "synchronized" which would result in us asking say HTML for completions for a change that they haven't seen yet.

Reply:
So if I understand you correctly, you're saying `LSPDocumentChangeKind.VirtualDocumentChanged` event fires sooner than `TextBuffer.PostChanged` and we don't want that when there are real changes. Is that right?

Reply:
Correct!